packages/@aws-cdk/aws-apprunner/lib/service.ts

```diff
@@ -0,0 +1,603 @@
+ * Describes the basic configuration needed for building and running an AWS App Runner service.
+ * This type doesn't support the full set of possible configuration options. Fur full configuration capabilities,
+ * use a `apprunner.yaml` file in the source code repository.
+ */
+export interface CodeConfigurationValues {
+  /**
+   * The command App Runner runs to build your application.
+   *
+   * @default - no build command.
+   */
+  readonly buildCommand?: string;
+  /**
+   * The port that your application listens to in the container.
+   *
+   * @default 8080
+   */
+  readonly port?: string;
+  /**
+   * A runtime environment type for building and running an App Runner service. It represents
+   * a programming language runtime.
+   */
+  readonly runtime: CodeRuntime;
+  /**
+   * The environment variables that are available to your running App Runner service.
+   *
+   * @default - no environment variables.
+   */
+  readonly environment?: { [key: string]: string };
+  /**
+   * The command App Runner runs to start your application.
+   *
+   * @default - no start command.
+   */
+  readonly startCommand?: string;
+}
+
+/**
+ * Represents the App Runner connection that enables the App Runner service to connect
+ * to a source repository. It's required for GitHub code repositories.
+ */
+export class Connection {
+  /**
+   * Using existing App Runner connection by specifying the connection ARN.
+   * @param arn connection ARN
+   * @returns Connection
+   */
+  public static fromConnectionArn(arn: string): Connection {
+    return new Connection(arn);
+  }
+  /**
+   * The ARN of the Connection for App Runner service to connect to the repository.
+   */
+  public readonly connectionArn: string
+  constructor(arn: string) {
+    this.connectionArn = arn;
+  }
+}
+
+/**
+ * Attributes for the App Runner Service
+ */
+export interface ServiceAttributes {
+  /**
+   * The name of the service.
+   */
+  readonly serviceName: string;
+  /**
+   * The ARN of the service.
+   */
+  readonly serviceArn: string;
+  /**
+   * The URL of the service.
+   */
+  readonly serviceUrl: string;
+  /**
+   * The status of the service.
+   */
+  readonly serviceStatus: string;
+}
+
+/**
+ * Represents the App Runner Service.
+ */
+export interface IService extends cdk.IResource {
+  /**
+   * The Name of the service.
+   */
+  readonly serviceName: string;
+  /**
+   * The ARN of the service.
+   */
+  readonly serviceArn: string;
+}
+
+/**
+ * The App Runner Service.
+ */
+export class Service extends cdk.Resource {
+  /**
+   * Import from service name.
+   */
+  public static fromServiceName(scope: Construct, id: string, serviceName: string): IService {
+    class Import extends cdk.Resource {
+      public serviceName = serviceName;
+      public serviceArn = cdk.Stack.of(this).formatArn({
+        resource: 'service',
+        service: 'apprunner',
+        resourceName: serviceName,
+      })
+    }
+    return new Import(scope, id);
+  }
+  /**
```

Comment:
Don't be shy with vertical spacing... 
```suggestion
  }

  /**
```